dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ranger/RangerConnector.java

```diff
@@ -226,22 +222,22 @@ protected String toCallDescription() {
     }
   }
 
-  private abstract static class AbstractRangerTask<T> extends AbstractTask<Void> {
+  private abstract static class AbstractRangerTask extends AbstractTask<Void> {
 
     public AbstractRangerTask(String targetPath) {
       super(targetPath);
     }
 
     @ForOverride
-    protected abstract Iterator<T> dataIterator(@Nonnull RangerClientHandle handle);
+    protected abstract Iterator<Object> dataIterator(@Nonnull RangerClientHandle handle);
 
     @Override
     protected Void doRun(TaskRunContext context, @Nonnull ByteSink sink, @Nonnull Handle handle)
         throws Exception {
       RangerClientHandle rangerClientHandler = (RangerClientHandle) handle;
       LOG.info("Writing to '{}' -> '{}'", getTargetPath(), sink);
       try (Writer writer = sink.asCharSink(StandardCharsets.UTF_8).openBufferedStream()) {
-        for (Iterator<T> iterator = dataIterator(rangerClientHandler); iterator.hasNext(); ) {
+        for (Iterator<Object> iterator = dataIterator(rangerClientHandler); iterator.hasNext(); ) {
```

Comment:
I looked briefly at this to see whether it _had_ to use Iterator, and technically, we could replace this with another type/class/pattern, even a RangerPagedCollection.forEach() pattern which would not require the exception transition. But that feels out of scope, so I'm leaving it here as an observation alone.

Reply:
The for loop body can throw `IOException` and `JsonProcessingException` - migrating to a `forEach` method (in fact, we could also reuse Iterator's `forEachRemaining`) would need another functional interface that mimics `Consumer` allowing it to throw, or adding try/catch exceptions in place.

Given that such a functional interface would be used only once and that the current form is quite simple to read, I prefer to keep it as is.